### PR TITLE
[Sensor] Use texture instead of render buffer for semantic info rendering

### PR DESCRIPTION
**Pull request by bigbike**
## Motivation and Context
As titled.
The benefit:
The texture is much easier to be visualized.

Should have no performance lose.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/esp/gfx/RenderTarget.cpp [hunk 1]
```diff
@@ -215,18 +223,23 @@ struct RenderTarget::Impl {
...
         "RenderTarget::Impl::readFrameObjectId(): this render target "
-        "was not created with objectId render buffer enabled.", );
-
-    framebuffer_.mapForRead(ObjectIdBuffer).read(framebuffer_.viewport(), view);
+        "was not created with objectId render texture enabled.", );
+#ifdef MAGNUM_TARGET_WEBGL
+    framebuffer_.mapForRead(ObjectIdTextureColorAttachment)
+        .read(framebuffer_.viewport(), view);
+#else
+    objectIdTexture_.image(0, view);
```
Comment (bigbike) [inline src/esp/gfx/RenderTarget.cpp:233]: @mosra: I used the `image(...)` here. Is it necessary?

**Review by erikwijmans — COMMENTED** [review]
I'm in principle fine with this.  I don't think there is any appreciable perf concerns but I will let @mosra comment as he knows this better than I.

**Comment by bigbike** [comment]
@erikwijmans : thanks. Pay attention to the cuda function. See if all the necessary changes are made.

File: src/esp/gfx/RenderTarget.cpp [hunk 2]
```diff
@@ -37,7 +38,7 @@ namespace gfx {
 
 const Mn::GL::Framebuffer::ColorAttachment RgbaBuffer =
     Mn::GL::Framebuffer::ColorAttachment{0};
-const Mn::GL::Framebuffer::ColorAttachment ObjectIdBuffer =
+const Mn::GL::Framebuffer::ColorAttachment ObjectIdTextureColorAttachment =
     Mn::GL::Framebuffer::ColorAttachment{1};
 const Mn::GL::Framebuffer::ColorAttachment UnprojectedDepthBuffer =
```
Comment (mosra) [inline src/esp/gfx/RenderTarget.cpp:43]: Why rename just one and not all? `ColorAttachment` seems like a clearer name, regardless of whether the underlying object is a renderbuffer or a texture.

File: src/esp/gfx/RenderTarget.h [hunk 3]
```diff
@@ -38,11 +38,11 @@ class RenderTarget {
...
-     * create objectId buffer
+     * create objectId texture
      * No need to set it for color sensor, depth sensor etc. as it makes the
      * rendering slower
      */
-    ObjectIdBuffer = 1 << 1,
+    ObjectIdTexture = 1 << 1,
     /**
      * @brief use depth texture, it must be set for the depth sensor.
      * No need to set it for color sensor, objectId sensor etc. as it makes the
```
Comment (mosra) [inline src/esp/gfx/RenderTarget.h:48]: Same here, rename them all to `*Attachment` to not need to differentiate between renderbuffers and textures?

File: src/esp/gfx/RenderTarget.cpp (hunk 1 above)
Comment (mosra) [reply src/esp/gfx/RenderTarget.cpp:233]: Why not have the same code path for both, like it was before? I don't think there should be any practical difference between the two, so in that case I'd say less code is better code :D

**Review by mosra — COMMENTED** [review]
> Should have no performance lose.

I know that's what I suggested based on my experience with native rendering. But just to be sure, did you (did anybody) verify this has no negative effect on the CUDA interop and WebGL performance? These are two cases where I'm not 100% certain.

File: src/esp/gfx/RenderTarget.cpp [hunk 4]
```diff
@@ -83,34 +88,37 @@ struct RenderTarget::Impl {
...
-    if (flags_ & Flag::RgbaBuffer) {
-      framebuffer_.mapForDraw({{0, RgbaBuffer}});
+    if (flags_ & Flag::RgbaAttachment) {
+      framebuffer_.mapForDraw(
+          {{Mn::Shaders::Generic3D::ColorOutput, RgbaBufferAttachment}});
     }
-
-    if (flags_ & Flag::ObjectIdBuffer) {
-      framebuffer_.mapForDraw({{1, ObjectIdBuffer}});
+    if (flags_ & Flag::ObjectIdAttachment) {
```
Comment (bigbike) [inline src/esp/gfx/RenderTarget.cpp:109]: @mosra:
There is a big bug hidden here that even you did not capture it.
When I need to visualize the semantic info, both `flags_ & Flag::RgbaAttachment` and `flags_ & Flag::ObjectIdAttachment` will be true.
That means `mapForDraw` will be called twice in two different if clauses. So the first attempt is overidden by the second. Sigh...

**Review by Skylion007 — APPROVED** [review]

**Pull request merged**